src/query/service/src/interpreters/interpreter_copy_v2.rs

```diff
@@ -52,6 +60,49 @@ impl CopyInterpreterV2 {
         Ok(CopyInterpreterV2 { ctx, plan })
     }
 
+    async fn filter_duplicate_files(
+        &self,
+        table_info: &StageTableInfo,
+        catalog_name: &str,
+        database_name: &String,
+        table_name: &String,
+        files: &[String],
+    ) -> Result<BTreeMap<String, TableStageFileInfo>> {
+        let req = GetTableStageFileReq {
+            table: TableNameIdent {
+                tenant: self.ctx.get_tenant(),
+                db_name: database_name.to_owned(),
+                table_name: table_name.to_owned(),
+            },
+            files: files.to_owned(),
+        };
+        let catalog = self.ctx.get_catalog(catalog_name)?;
+        let resp = catalog.get_table_stage_file_info(req).await?;
+        let mut file_map = BTreeMap::new();
+
+        for file in files.iter() {
+            let stage_file = list_files(&self.ctx, &table_info.stage_info, file, "").await?;
+            if stage_file.is_empty() {
+                continue;
+            }
+
+            let stage_file = &stage_file[0];
+
+            if let Some(file_info) = resp.file_info.get(file) {
+                // No need to copy the file again only if md5 match.
+                if file_info.md5 == stage_file.md5 {
```

Comment:
We need to check the following things:

- name
- length
- last modified
- etag (use etag instead of md5)

Reply:
> if the location is https(not s3), the etag is?

ETag in OpenDAL is the same `ETag` defined in HTTP Standard: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag

Reply:
the type `StageFile`, which save stage file meta info, has no `etag` field:

```
#[derive(Default, Clone)]
pub struct StageFile {
    pub path: String,
    pub size: u64,
    pub md5: Option<String>,
    pub last_modified: DateTime<Utc>,
    pub creator: Option<UserIdentity>,
}
```